PageObject/Config/WebdriverSetup.py

```diff
@@ -2,11 +2,24 @@
 from selenium.webdriver.chrome.options import Options
 
 
-def webdriver_setup():
-    chrome_options = Options()
-    chrome_options.add_experimental_option("detach", True)
-    driver = webdriver.Chrome(options=chrome_options)
-
-    driver.get("https://www.lambdatest.com/")
-    driver.maximize_window()
-    return driver
+# def webdriver_setup():
+#     chrome_options = Options()
+#     chrome_options.add_experimental_option("detach", True)
+#     driver = webdriver.Chrome(options=chrome_options)
+#
+#     driver.get("https://www.lambdatest.com/")
+#     driver.maximize_window()
+#     return driver
+
+class WebdriverSetup:
+    def __init__(self, driver):
```

Comment:
ogolnie driver generujesz dopiero webdriver_setup  wiec nie powinineś nic do inita przekazywać ( wtym przypadku nawet init ci nie jest potrzebny)
```suggestion
class WebdriverSetup:
   def webdriver_setup(self):
        chrome_options = Options()
        chrome_options.add_experimental_option("detach", True)
        driver = webdriver.Chrome(options=chrome_options)
        driver.get("https://www.lambdatest.com/")
       driver.maximize_window()
       return driver
```

Reply:
mozesz ewntualnie zrobic
```suggestion
class WebdriverSetup:
    def __init__(self):
       self.driver = self._get_driver()

    def _get_driver(self):
        chrome_options = Options()
        chrome_options.add_experimental_option("detach", True)
        driver = webdriver.Chrome(options=chrome_options)
        driver.get("https://www.lambdatest.com/")
        driver.maximize_window()
        return driver
    
```

Reply:
i wtedy bys wołał

web_driver_client = WebdriverSetup()
web_driver_client.driver()


albo w tym pierwsyzm wypadku
WebdriverSetup.webdriver_setup()   tylko wtedy zrób metode jako statyczną